flocker/node/script.py

```diff
@@ -179,11 +132,47 @@ def get_service(self, reactor, options):
         """
         return AgentLoopService(
             reactor=reactor,
-            deployer=self.deployer_factory(hostname=options["hostname"]),
+            deployer=self.deployer_factory(options=options),
```

Comment:
So this should parse the config-file, and pass the options relevant to the deployer into the factory.